tmt/steps/prepare/feature.py

```diff
@@ -0,0 +1,280 @@
+import dataclasses
+import enum
+import re
+from pathlib import Path
+from typing import Optional, cast
+
+import tmt
+import tmt.base
+import tmt.log
+import tmt.options
+import tmt.steps
+import tmt.steps.prepare
+import tmt.utils
+from tmt.steps.provision import Guest
+from tmt.utils import ShellScript, field
+
+
+class Distro(enum.Enum):
```

Comment:
Nice progress!

Now, the question is, do we really need this enum, and the code "guessing" the guest distro? Because Ansible has its own powerful fact discovery, and together with `when` and `block`, we can offload this into playbooks and end up with just `epel-enable.yaml` and `epel-disable.yaml` :)

Let's say the `epel-enable.yaml` would turn into this:

```yaml
- hosts: guest
  user: root
  tasks:
    - name: Install package 'epel-release'
      ansible.builtin.yum:
        name: "https://dl.fedoraproject.org/pub/epel/epel-release-latest-{{ ansible_distribution_major_version }}.noarch.rpm"
        state: present

    - when: ansible_distribution_major_version == "7"
      block:
        - name: Install package 'yum-utils'
          ansible.builtin.yum:
            name: yum-utils
            state: present

        - name: Enable repos related to 'epel'
          shell: yum-config-manager --enable epel epel-debuginfo epel-source

    - when: ansible_distribution_major_version == "8" or ansible_distribution_major_version == "9"
      block:
        - name: Install package 'epel-next-release'
          ansible.builtin.yum:
            name: "https://dl.fedoraproject.org/pub/epel/epel-next-release-latest- 
   {{ ansible_distribution_major_version }}.noarch.rpm"
            disable_gpg_check: true
            state: present

        - name: Install 'dnf config-manager'
          shell: dnf -y install 'dnf-command(config-manager)'

        - name: Enable 'epel' repos
          shell: dnf config-manager --enable epel epel-debuginfo epel-source

        - name: Enable 'epel-next' repos
          shell: dnf config-manager --enable epel-next epel-next-debuginfo epel-next-source
```

I believe we can also expose our current guest facts to playbooks if needed, to provide extra info, but for distro detection, what Ansible already offers should be enough. What do you think?